The size might be a long long, so deal with that.

--- krb5-1.3/src/appl/gssftp/ftpd/ftpcmd.y
+++ krb5-1.3/src/appl/gssftp/ftpd/ftpcmd.y
@@ -1515,12 +1515,12 @@
 		    (stbuf.st_mode&S_IFMT) != S_IFREG)
 			reply(550, "%s: not a plain file.", filename);
 		else
-			reply(213, "%lu", (long) stbuf.st_size);
+			reply(213, "%llu", (long long) stbuf.st_size);
 		break;}
 	case TYPE_A: {
 		FILE *fin;
 		register int c;
-		register long count;
+		register long long count;
 		struct stat stbuf;
 		fin = fopen(filename, "r");
 		if (fin == NULL) {
@@ -1542,7 +1542,7 @@
 		}
 		(void) fclose(fin);
 
-		reply(213, "%ld", count);
+		reply(213, "%lld", count);
 		break;}
 	default:
 		reply(504, "SIZE not implemented for Type %c.", "?AEIL"[type]);
